public/app/app.ts

```diff
@@ -206,24 +207,19 @@ export class GrafanaApp {
       setDataSourceSrv(dataSourceSrv);
       initWindowRuntime();
 
-      let preloadResults: PluginPreloadResult[] = [];
+      // Initialize plugin extensions
+      const extensionsRegistry = new ReactivePluginExtensionsRegistry();
+      extensionsRegistry.register({
+        pluginId: 'grafana',
+        extensionConfigs: getCoreExtensionConfigurations(),
+      });
 
       if (contextSrv.user.orgRole !== '') {
-        // Preload selected app plugins
-        preloadResults = await preloadPlugins(config.apps);
+        preloadPlugins(config.apps, extensionsRegistry);
```

Comment:
We should only do this on demand when a plugin that has a specific extension id is needed.

Loading all plugins at app boot for all pages is still very problematic (any global bug/side effect will impact all pages)

Reply:
@torkelo Yea, that's our goal as well, however we need these changes to the registry before we could get there. As this PR brings improvements compared to where we are today (we are currently loading all plugins, too, but right now it is blocking the grafana app init as well), I think it would make sense to merge it until we have the correct pre-loading behaviour in place.

Regarding preloading plugins only when they are needed: I don't think we are too far from that, [we are working on a tool that can extract meta-information from plugins automatically](https://github.com/grafana/plugin-tools/pull/805), which we will be able to use to pre-load a plugin only when it is needed.